src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -266,19 +270,37 @@ private static void warn(String message) {
     return matches;
   }
 
-  // TODO: docs
-  private static List<GenomeRange> generateBEDFromVCF(VCFFileReader vcfReader) {
-    List<GenomeRange> toReturn = new ArrayList<>();
+  /**
+   * Generates set of genome ranges from the variants in the VCF file
+   *
+   * For each variant in the input VCF, it will create a region corresponding to the this variant
+   * padded with VARIANT_CONTEXT_SIZE bps from each side.
+   *
+   * @param vcfReader the input VCF file reader
+   * @return a hashmap of chromosome to arraylist of GenomeRanges corresponsing to variants in the inpit VCF
+   */
+  private static Map<String, List<GenomeRange>> generateBEDFromVCF(VCFFileReader vcfReader) {
```

Comment:
By padding immediately upon loading of the VCF records the downstream logic is complicated. E.g. imagine regions

chr1   0   9
chr1  12  20

and a variant at chr1   10    11. We don't want to convert this variant as it's actually not in the confident region. But due to padding the query regions will get artificially broken up by the flanks of this variant.

The cleaner conceptual algorithm here is to create a BED file from the VCF that includes no padding. Retain only those variant BED records that are completely encompassed by a confident region, and pad those based on the requested padding and the boundaries of the encompassing region. Then the merged BED logic can be simplified further too.

Reply:
Done.